pandas/tests/groupby/test_grouping.py

```diff
@@ -628,6 +628,26 @@ def test_groupby_empty(self):
         # check name
         assert s.groupby(s).grouper.names == ["name"]
 
+    def test_groupby_level_index_value_all_na(self):
+        # issue 20519
+        df = pd.DataFrame(
+            [["x", np.nan, 10], [None, np.nan, 20]], columns=["A", "B", "C"]
```

Comment:
If I cast `df["B"] = df["B"].astype("datetime64")` before doing the `set_index`, I get a different error on the groupby call (in master).  Should that be fixed by this PR?  If so, please test.

side-note, I'd find this easier to follow in smaller steps:

```
df = pd.DataFrame(...)
df = df.set_index(["A", "B"])
gb = df.groupby(level=["A", "B"])
result = gb.sum()
```

Especially relevant as it is the `gb = ...` line that raises in master

Reply:
@jbrockmendel It's different issue. Because if `df["B"]` are all NaN, groupby could not raise exception. This is also bug. but i think, this should be fixed before when all NaN value column could be groupby. What i changed is about MultiIndex.

Reply:
@jbrockmendel 
I'm bit confused. you `df["B"] = df["B"].astype("datetime64")` means `df["B"] = df["B"].astype("datetime64[ns]")` write?
in master, 
`df = pd.DataFrame(...)`
`df["B"] = df["B"].astype("datetime64[ns]")`
`df = df.set_index(["A", "B"])`
`gb = df.groupby(level=["A", "B"])`
and 
`df = pd.DataFrame(...)`
`df = df.set_index(["A", "B"])`
`gb = df.groupby(level=["A", "B"])`
raise same IndexError :cannot do a non-empty take from an empty axes.

Reply:
@jbrockmendel
I'm bit confused. you df["B"] = df["B"].astype("datetime64") means df["B"] = df["B"].astype("datetime64[ns]") write?
in master,
df = pd.DataFrame(...)
df["B"] = df["B"].astype("datetime64[ns]")
df = df.set_index(["A", "B"])
gb = df.groupby(level=["A", "B"])
and
df = pd.DataFrame(...)
df = df.set_index(["A", "B"])
gb = df.groupby(level=["A", "B"])
raise same IndexError :cannot do a non-empty take from an empty axes.

Reply:
> I'm bit confused. you df["B"] = df["B"].astype("datetime64") means df["B"] = df["B"].astype("datetime64[ns]") write?

Yes, I meant to write `"datetime64[ns]"` and not just `"datetime64"`.

> raise same IndexError

Huh, not sure how I got to a different error.  My bad.